pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1685,7 +1689,18 @@ func (ctrl *PersistentVolumeController) findDeletablePlugin(volume *v1.Persisten
 		if provisionPluginName != "" {
 			plugin, err := ctrl.volumePluginMgr.FindDeletablePluginByName(provisionPluginName)
 			if err != nil {
-				if !strings.HasPrefix(provisionPluginName, "kubernetes.io/") {
+				if strings.HasPrefix(provisionPluginName, "kubernetes.io/") {
+					if ctrl.csiMigratedPluginManager.IsCSIMigrationEnabledForPluginByName(provisionPluginName) {
```

Comment:
Is this change related to this PR or is it a fix for the deletion thing? If the latter please do this change in a seperate PR

Reply:
It is for the deletion thing. Will move it out. Thanks for catching it!

Reply:
is there another PR that has this fix? could you link that here (want to make sure it does get fixed)

Reply:
That issue is being tracked at https://github.com/kubernetes/kubernetes/issues/79043 with notes on the changes necessary in-tree as well in external provisioner to prevent potential races.